care/facility/api/serializers/patient.py

```diff
@@ -105,8 +113,24 @@ def validate_facility(self, value):
 
     def validate(self, attrs):
         validated = super().validate(attrs)
-        if not self.partial and not validated.get("age") and not validated.get("date_of_birth"):
-            raise serializers.ValidationError({"non_field_errors": [f"Either age or date_of_birth should be passed"]})
+        # flag to check whether validation failed
+        err_flag = False
+        errors = ""
+        date_of_birth = validated.get("date_of_birth")
+        if not self.partial and not validated.get("age") and not date_of_birth:
```

Comment:
Need one more change here. Make `date_of_birth` as a required field. Age is not required.